### DISCUSSION TITLE
Is the passport file supposed to be named, '[...auth].ts' ?

### DISCUSSION THREAD

**Discussion by insanity54** [blitz-js/blitz#2129] [Q&A] [answered]
I'm a bit confused about the tutorial [Third Party Login with Passport](https://blitzjs.com/docs/passportjs). The tutorial says the following.

> Add a new api route at `app/api/auth/[...auth].ts` with the following contents.

I see that strange filename and it makes me wonder... Is the tutorial telling me to specifically use the filename, `[...auth].ts` or is it using javascript spread syntax used to hint at what the filename should be?

For example, if I'm implementing a Twitter login, should I be creating the file, 'app/api/auth/twitter.js'?

**Comment by Skn0tt** [comment, marked as answer]
`api/auth/[...auth].ts` is indeed the correct filename. It's a so-called [dynamic API Route](https://blitzjs.com/docs/api-routes#dynamic-api-routes), which basically matches all routes that start with `auth/`.
